ophyd/controls/signal.py

```diff
@@ -423,20 +450,25 @@ def describe(self):
         dict
             Dictionary of name and formatted description string
         """
-        return {self.name: {'source': 'PV:{}'.format(self._read_pv.pvname),
-                            'dtype': 'number',
+        return {self.name: {'source': 'PV:%s' % self._read_pv.pvname,
+                            'dtype': self.dtype,
+                            'formatter': self.formatter,
                             'shape': []}}
 
     def read(self):
-        """Read the signal and format for data collection
+        """Read the signal and combine it with its timestamp.
+
+        This value is formatted according to the values in self.dtype and
+        self.format
 
         Returns
         -------
         dict
             Dictionary of value timestamp pairs
+            {'value': value, 'timestamp': timestamp}
         """
-
-        return {self.name: {'value': self.value,
+        value = self.describe()['dtype'](self.formatter % self.value)
```

Comment:
If we are going to round for this purpose we should be using http://docs.scipy.org/doc/numpy/reference/generated/numpy.round_.html which lets you set how many decimals to keep and I know that the numpy folks are careful to make sure the rounding does not introduce any bias, I am not sure that string formatting does (I know that the built-in python `ronud` is biased).